File: Data-Agent-Server/data-agent-server-app/src/main/java/edu/zsc/ai/domain/service/db/TableService.java
package edu.zsc.ai.domain.service.db;

import edu.zsc.ai.domain.model.dto.response.db.TableDataResponse;

import java.util.List;

public interface TableService {

    List<String> listTables(Long connectionId, String catalog, String schema, Long userId);

    String getTableDdl(Long connectionId, String catalog, String schema, String tableName, Long userId);

    void deleteTable(Long connectionId, String catalog, String schema, String tableName, Long userId);

    TableDataResponse getTableData(Long connectionId, String catalog, String schema, String tableName, Long userId, Integer currentPage, Integer pageSize);
}

File: Data-Agent-Client/src/services/table.service.ts
import http from '../lib/http';
import { ApiPaths } from '../constants/apiPaths';

export const tableService = {
  listTables: async (connectionId: string, catalog?: string, schema?: string): Promise<string[]> => {
    const params: Record<string, string> = { connectionId };
    if (catalog != null && catalog !== '') params.catalog = catalog;
    if (schema != null && schema !== '') params.schema = schema;
    
    const response = await http.get<string[]>(ApiPaths.TABLES, { params });
    return response.data;
  },

  getTableDdl: async (
    connectionId: string,
    tableName: string,
    catalog?: string,
    schema?: string
  ): Promise<string> => {
    const params: Record<string, string> = {
      connectionId,
      tableName
    };
    if (catalog != null && catalog !== '') params.catalog = catalog;
    if (schema != null && schema !== '') params.schema = schema;

    const response = await http.get<string>(ApiPaths.TABLES_DDL, { params });
    return response.data;
  },

  deleteTable: async (
    connectionId: string,
    tableName: string,
    catalog?: string,
    schema?: string
  ): Promise<void> => {
    await http.delete(ApiPaths.TABLES, {
      data: {
        connectionId,
        tableName,
        catalog,
        schema
      }
    });
  },
};
